### ISSUE TITLE
Metric on the health status of RPCs in the pool.

### ISSUE THREAD

**Issue by FranckUltima** [FTKuhnsman/rpc-balancer#2]
Is there or would it be possible to add a metric to display in Grafana either the RPC currently being used on the RPC balancer, or the current health status of all RPCs registered in the balancer?

**Comment by FTKuhnsman** [comment]
> Is there or would it be possible to add a metric to display in Grafana either the RPC currently being used on the RPC balancer, or the current health status of all RPCs registered in the balancer?

I could add a health metric with each node as a separate label. There are some current limitations though... Health checks are only performed on unhealthy nodes. A node can only go unhealthy if a request is made to it and fails. Therefore, in failover mode, all nodes will show as healthy until the primary node has an error, then the secondary, and so on... It only checks unhealthy nodes to avoid frequent health checks against free rpc endpoints so there are no issues with rate limiting.

Displaying the current RPC being used depends on the selection method. In failover mode this would make sense, but in random or roundrobin it would also be using a different node.

**Comment by FranckUltima** [comment]
It would already be great to simply be able to trace back in Grafana which RPC is active in live mode in failover mode. 
This would allow us to know that an RPC is no longer operational. Currently, I know that my Arbitrum node is down if I see in my Grafana that my Livepeer nodes are no longer synchronizing. 
With failover, I no longer see this. So being able to know the RPC used in live would allow me to keep this information and know that if I do maintenance on my node, the Livepeer nodes have switched to a failover RPC.

**Comment by FranckUltima** [comment]
For your information, if it may be helpful and if I am not mistaken, to avoid exposing the RPC ports and since Docker bypasses UFW rules, I have limited the exposed ports to the local network:

ports:
  - 127.0.0.1:8080:8080
  - 127.0.0.1:8081:8081
If this can be useful to specify for users like me who are less used to Docker.

**Comment by FTKuhnsman** [comment]
I have changed the health metric to have labels representing each endpoint. keep in mind the health metric is only updated when a node fails and is later updated via the health check. health checks are only performed for unhealthy nodes, so in failover mode all endpoints default to healthy until one fails. they will only be updated to "unhealthy" in order (i.e., domino effect). round robin and random selection options will identify potentially unhealth nodes faster.

I also updated the docs so the docker-compose example restricts access to 127.0.0.1
